Restructure extension code and add fixes for Windows paths and views not loading

SanjulaGanepola opened the pull request:
### Changes
- [x] Major restructuring of files
- [x] Update extension details in `package.json`
- [x] Removed `vscode-sourceorbit:projectsLoaded` context in favor of having views always shown with welcome view text
- [x] Fix windows path issue in `getImpacts` request in server code
- [x] Fix views not rendering anything if project not loaded
- [x] Fix affected object view not showing anything when extension first loaded and file is open

I have more changes to make, but will add them in separate smaller PRs

---

Note added by @worksofliam: this only affects `vs/`

worksofliam commented:
@SanjulaGanepola I can get this review started on Monday. Fab work!

SanjulaGanepola commented on vs/client/src/views/projectExplorer/ileObjectTreeItem.ts:
```diff
@@ -0,0 +1,31 @@
+export class ILEObjectTreeItem extends TreeItem implements ProjectExplorerTreeItem {
+	constructor(public workspaceFolder: WorkspaceFolder, private ileObject: ILEObject, canExpand = false) {
+		super(`${ileObject.systemName}.${ileObject.type}`, canExpand ? TreeItemCollapsibleState.Collapsed : TreeItemCollapsibleState.None);
+		// const logs = TargetsManager.getLogs(workspaceFolder, ileObject);
```
@worksofliam Are these logs something that we should render? Maybe in the tooltip?

worksofliam commented:
@SanjulaGanepola which repos are you testing this with? ibmi-company_system and bob-recursive-example?

worksofliam commented:
@SanjulaGanepola This view under source control is always showing the welcome page. This is for company_system. Any clues?

<img width="363" alt="image" src="https://github.com/user-attachments/assets/2a379579-c258-4a98-825e-0a9d4ef93c9c">

SanjulaGanepola commented:
@worksofliam That is odd cuz it appears fine for me. Does the active impact view work? The only real change I made to this logic was https://github.com/IBM/sourceorbit/pull/83/files#diff-5a23ea1f11048195dc7ad0fdf0d21e5b66e20339b68c5f1a7e42a1ccef0fe085R28

![image](https://github.com/user-attachments/assets/21d73a55-c8fe-4ade-a450-e5813ca7f5de)

worksofliam commented:
@SanjulaGanepola Yup, it works! But, I think we need to give the two 'Affected Objects' windows need to have different names (which is totally my bad, because I think I wrote that :joy:)

Perhaps the 'Affected Objects' under Explorer should be 'Source Effects' and the one under Source Control should be 'Affected Objects' (no change).

What do you think?

SanjulaGanepola commented:
@worksofliam Oh wait, were you testing with the company_system test fixture. I was using the actual https://github.com/IBM/ibmi-company_system project. Since we have that in our .gitignore, I think it makes sense we don't show any Git related impacts.

I did remove it from our .gitignore to test it out and did find another bug though with [this logic](https://github.com/IBM/sourceorbit/blob/main/vs/client/src/git/index.ts#L28) not detecting the repo. I assume we want to support opening any folder within the project being opened as a workspace folder. I will push a fix for this.

> we need to give the two 'Affected Objects' windows different names

Totally agree. I think Edmund's suggestion makes sense for the names. I will push this change too.

SanjulaGanepola commented:
Nevermind...you are using ibmi-company-system. Also forgot your source control view literally has detected changes 😅. Still investigating...

SanjulaGanepola commented:
> I did remove it from our .gitignore to test it out and did find another bug though with [this logic](https://github.com/IBM/sourceorbit/blob/main/vs/client/src/git/index.ts#L28) not detecting the repo. I assume we want to support opening any folder within the project being opened as a workspace folder. I will push a fix for this.

@worksofliam Ok so that part can be fixed by changing:
```ts
const repository = gitApi.repositories.find(r => r.rootUri.fsPath === workspaceFolder.uri.fsPath);
```
to:
```ts
const repository = gitApi.getRepository(workspaceFolder.uri);
```

but problems could arise [here](https://github.com/IBM/sourceorbit/blob/refactor/source-orbit-tree-item/vs/client/src/views/impactView/impactView.ts#L27), if there are changes in the project, but they are outside the current workspace folder as a sub folder is opened. Do we need to support opening sub folders?

worksofliam commented:
@SanjulaGanepola In theory no sub folders, just multiple workspace uris.

SanjulaGanepola commented:
> This view under source control is always showing the welcome page.

@worksofliam I can't quite tell what broke this since I can't reproduce it. Could you debug [this spot](https://github.com/IBM/sourceorbit/pull/83/files#diff-5a23ea1f11048195dc7ad0fdf0d21e5b66e20339b68c5f1a7e42a1ccef0fe085R28) and share any findings?

worksofliam commented:
@SanjulaGanepola for the record, I don't think anything is wrong with the code. I think there was one instance when the git extension did not trigger the change event and so our view didn't update. If I manually refresh, or change a file, the view updates.

worksofliam commented:
@SanjulaGanepola I've got this lint rule being throw in `api.ts`:

<img width="826" alt="image" src="https://github.com/user-attachments/assets/e9989b11-0755-4f05-9b88-f9317453f4cf">

worksofliam commented on vs/client/src/languageClientManager.ts:
```diff
@@ -0,0 +1,120 @@
+		if (LanguageClientManager.client) {
+			return LanguageClientManager.client.sendRequest(`reloadProject`, [workspaceFolder.uri.toString()]);
+		} else {
+			return Promise.reject()
```
Missing semi colon.

worksofliam commented on vs/client/src/tasks.ts:
```diff
@@ -1,58 +1,60 @@
-interface SourceOrbitTask extends TaskDefinition {
-	builder: "bob"|"make"|"json";
-}
+export namespace SourceOrbitTask {
```
Same namespace issue here.

SanjulaGanepola commented:
@worksofliam Addressed the comments. I was in favor of disabling that specific lint error so we can use namespaces. Lmk if you think we should revert this.

worksofliam commented:
@SanjulaGanepola i am okay with disabled that rule, though only because we don't really have an exportable API. If we do in the future, we should enable it again.

I will review one more time tomorrow and likely merge it now. Then I'll start the next PR - thanks for your work here!
